### PR TITLE
DOC: corrected docstring for interpolate.griddata

### LINKED ISSUE #4802: griddata docstring doesn't make any sense
The docstring for `griddata` refers to many different integers:

`D`, `n`, `ndim`, `M`:

```
Interpolate unstructured D-dimensional data.

Parameters
----------
points : ndarray of floats, shape (n, D)
    Data point coordinates. Can either be an array of
    shape (n, D), or a tuple of `ndim` arrays.
values : ndarray of float or complex, shape (n,)
    Data values.
xi : ndarray of float, shape (M, D)
    Points at which to interpolate data.
```

The only one that is defined is `D`, so users have no idea what the others are.

Furthermore the example provided contradicts the docstring as xi is passed as a tuple but the docstring says it must be a ndarray.

### PR DESCRIPTION
**Pull request by aypiggott**
Clarified usage and fixed description to match actual implementation of griddata. Fixes #4802.

### CODE REVIEW

**Comment by aypiggott** [comment]
By the way, this is my first pull request to scipy - please let me know if there's anything I should change. Thanks!

File: scipy/interpolate/ndgriddata.py [hunk 1]
```diff
@@ -92,13 +92,14 @@ def griddata(points, values, xi, method='linear', fill_value=np.nan,
 
     Parameters
     ----------
-    points : ndarray of floats, shape (n, D)
-        Data point coordinates. Can either be an array of
-        shape (n, D), or a tuple of `ndim` arrays.
-    values : ndarray of float or complex, shape (n,)
+    points : 2-D ndarray of floats with shape (n, D), or tuple of 1-D ndarrays
+        Data point coordinates. Can either be a 2-D ndarray with
+        shape (n, D), or a length D tuple of 1-D ndarrays with shape (n,).
```
Comment (thisch) [inline scipy/interpolate/ndgriddata.py]: Isn't the 2nd sentence redundantl?

File: scipy/interpolate/ndgriddata.py [hunk 2]
```diff
@@ -92,13 +92,14 @@ def griddata(points, values, xi, method='linear', fill_value=np.nan,
...
+    values : 1-D ndarray of float or complex with shape (n,)
         Data values.
-    xi : 2-D ndarray of float or tuple of 1-D array, shape (M, D)
-        Points at which to interpolate data.
+    xi : 2-D ndarray of floats with shape (m, D), or tuple of ndarrays
+        Points at which to interpolate data. Can either be a 2-D ndarray with
+        shape (m, D), or a length D tuple of ndarrays with identical dimensions.
```
Comment (thisch) [inline scipy/interpolate/ndgriddata.py]: Same here

**Review by twmr — CHANGES_REQUESTED** [review]

**Comment by aypiggott** [comment]
Ok, as suggested I've removed the redundant second sentences.

File: scipy/interpolate/ndgriddata.py [hunk 3]
```diff
@@ -92,12 +92,11 @@ def griddata(points, values, xi, method='linear', fill_value=np.nan,
 
     Parameters
     ----------
-    points : ndarray of floats, shape (n, D)
-        Data point coordinates. Can either be an array of
-        shape (n, D), or a tuple of `ndim` arrays.
-    values : ndarray of float or complex, shape (n,)
+    points : 2-D ndarray of floats with shape (n, D), or length D tuple of 1-D ndarrays with shape (n,).
+        Data point coordinates. 
+    values : 1-D ndarray of float or complex with shape (n,)
```
Comment (pv) [inline scipy/interpolate/ndgriddata.py]: It doesn't have to be 1D, for `ndim` input each data point is a `ndim-1` array.

File: scipy/interpolate/ndgriddata.py [hunk 4]
```diff
@@ -92,12 +92,11 @@ def griddata(points, values, xi, method='linear', fill_value=np.nan,
...
         Data values.
-    xi : 2-D ndarray of float or tuple of 1-D array, shape (M, D)
+    xi : 2-D ndarray of floats with shape (m, D), or length D tuple of ndarrays with identical shapes.
```
Comment (pv) [inline scipy/interpolate/ndgriddata.py]: shapes don't need to be identical, broadcastable to the same shape is enough

**Comment by rgommers** [comment]
@aypiggott can you please address the last comments of @pv so we can merge this?

File: scipy/interpolate/ndgriddata.py (hunk 3 above)
Comment (vyasr) [reply scipy/interpolate/ndgriddata.py]: I can bring this PR up-to-date, but I'm a little confused about this comment. The set of points is a 2D array of all points, which can each be D dimensional. So the values should be a 1 dimensional array of length n, correct? Why is this wrong?

**Comment by BinSquare** [comment]
I'm interested in addressing the last comment by pv @rgommers and finishing the stalled pull request.

**Comment by clementkng** [comment]
@rgommers @pv I've addressed the feedback [here](https://github.com/scipy/scipy/pull/10340).

**Comment by pv** [comment]
done in #10340

**Pull request closed without merging**